hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/MergeNormalizationPlan.java

```diff
@@ -78,4 +82,30 @@ public void execute(Admin admin) {
       LOG.error("Error during region merge: ", ex);
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    MergeNormalizationPlan that = (MergeNormalizationPlan) o;
+
+    return new EqualsBuilder()
+      .append(firstRegion, that.firstRegion)
+      .append(secondRegion, that.secondRegion)
+      .isEquals();
+  }
+
+  @Override
+  public int hashCode() {
+    return new HashCodeBuilder(17, 37)
```

Comment:
wtb generate the hashCode at the beginning (in constructor)? instead of calculating every time.

Reply:
the same below

Reply:
Nice one, since we have two `final RegionInfo` and hopefully they are not getting mutated after plan generation, we can do this.

Reply:
And make the object instances that much larger? Is that really helpful? I think of this as a premature optimization, something the JIT can handle for me if it thinks so.

If you feel strongly about it, I suppose...

Reply:
I don't have strong opinion since I have also never done it, just that I felt what if this is beneficial but it seems it is not proper optimization. It's fine to keep it as is in that case.
@Reidddddd what do you think? Not a major improvement right?

Reply:
Step back, I don't think it ever needs hashCode() method, they are currently stored in ArrayList. Why not implement it when needed.

Reply:
So what I really wanted was `equals`, so that I could treat these objects as POJOs and do simple comparisons in unit tests. In java, it's a bad idea to implement a custom `equals` without also implementing `hashCode`, so I did both. Well, I generated both from IntelliJ without much thought about it.

Reply:
please make sure we implement `hashCode` if we implement `equals`.

if we're not expecting to call `hashCode` then why are we going through ways to optimize the call?

Reply:
> please make sure we implement hashCode if we implement equals

 `hashCode` and `equals` are not binding. `hashCode` only makes sense when object will be used as a key in `Map` or as en element in `Set`, with one rare case: `hashCode` is used as a part of judging `equals`.

> then why are we going through ways to optimize the call?

Because I lacked the context. After I read the codes in trunk and got the intention of this PR, then found out it unnecessary.

Reply:
From [javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)) on `Object#equals(Object)`

> Note that it is generally necessary to override the `hashCode` method whenever this method is overridden, so as to maintain the general contract for the `hashCode` method, which states that equal objects must have equal hash codes.

And besides since I want to treat these objects like POJOs, it is useful to be able to use them in maps and sets.